google-cloud-firestore/lib/google/cloud/firestore/transaction.rb

```diff
@@ -102,18 +113,41 @@ def firestore
         #     end
         #   end
         #
-        def get_all *docs
+        # @example Get docs using a field mask:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   firestore.transaction do |tx|
+        #     # Get and print city documents
+        #     cities = ["cities/NYC", "cities/SF", "cities/LA"]
+        #     tx.get_all(*cities, field_mask: :population).each do |city|
+        #       puts "#{city.document_id} has #{city[:population]} residents."
+        #     end
+        #   end
+        #
+        def get_all *docs, field_mask: nil
           ensure_not_closed!
           ensure_service!
 
-          return enum_for(:get_all, docs) unless block_given?
+          unless block_given?
+            return enum_for(:get_all, docs, field_mask: field_mask)
+          end
 
           doc_paths = Array(docs).flatten.map do |doc_path|
             coalesce_doc_path_argument doc_path
           end
+          mask = Array(field_mask).map do |field_path|
+            if field_path.is_a? FieldPath
+              field_path.formatted_string
+            else
+              FieldPath.parse(field_path).formatted_string
+            end
+          end
+          mask = nil if mask.empty?
```

Comment:
Refactor this (and the similar code in client.rb) to `Convert` to remove duplication?

Reply:
I don't think this benefits that much from a well named abstraction. I considered it and decided to leave it inline, matching other existing instances of these calls in the code.

```ruby
# Compare:
FieldPath.parse(field_path).formatted_string
# to:
Convert.field_path_to_formatted_string(field_path)
```